lib/MusicBrainz/Server/Data/Area.pm

```diff
@@ -66,100 +70,57 @@ sub load
     load_subobjects($self, ['area', 'begin_area', 'end_area', 'country'], @objs);
 }
 
-sub load_containment
-{
+sub get_containment_query {
+    my ($self, $link_type_param, $descendant_param) = @_;
+
+    return (qq{
+        WITH RECURSIVE area_descendants AS (
+            SELECT entity0 AS parent, entity1 AS descendant, 1 AS depth
+              FROM l_area_area laa
+              JOIN link ON laa.link = link.id
+             WHERE link.link_type = $link_type_param
+               AND entity1 = $descendant_param
```

Comment:
This `AND entity1 = ?` bit is the critical part which was added and makes the recursive query much faster.